encfs/CipherFileIO.cpp

```diff
@@ -383,26 +383,73 @@ int CipherFileIO::generateReverseHeader(unsigned char *headerBuf) {
  * Read block from backing plaintext file, then encrypt it (reverse mode)
  */
 ssize_t CipherFileIO::readOneBlock(const IORequest &req) const {
+
+  int cbs = fsConfig->cipher->cipherBlockSize();
   int bs = blockSize();
   off_t blockNum = req.offset / bs;
 
   IORequest tmpReq = req;
 
+  // adjust offset and length if we pad
+  if (haveCBCPadding && !fsConfig->reverseEncryption) {
+    tmpReq.offset += tmpReq.offset / tmpReq.dataLen;
+    tmpReq.dataLen += 1;
+  }
+
```

Comment:
You use the remainder of the division of the offset by the request length.... Why? What does this do?

Reply:
Here, `tmpReq.dataLen == blockSize` (each read / write request is `blockSize` bytes in length).
Remember that every block is padded with 1 byte.
We then add to the requested offset the number of backing padding bytes which exist before the upper position we want to read from.